sympy/tensor/indexed.py

```diff
@@ -137,7 +139,19 @@ class Indexed(Expr):
     is_symbol = True
     is_Atom = True
 
-    def __new__(cls, base, *args, **kw_args):
+    @staticmethod
+    def _set_assumptions(obj, assumptions=None):
+        """Set assumptions on obj, making sure to apply consitent values."""
+        assumptions = dict() if assumptions is None else assumptions
+        Symbol._sanitize(assumptions)
+        tmp_asm_copy = assumptions.copy()
+
+        is_commutative = fuzzy_bool(assumptions.get('commutative', True))
+        assumptions['commutative'] = is_commutative
+        obj._assumptions = StdFactKB(assumptions)
+        obj._assumptions._generator = tmp_asm_copy  # Issue #8873
+
+    def __new__(cls, base, *args, assumptions=None, **kw_args):
```

Comment:
This is not valid syntax in Python 2. You need to use `kw_args.setdefault('assumptions', None)`.